mcs/class/System/System.Diagnostics/Win32EventLog.cs

```diff
@@ -703,45 +702,52 @@ private IntPtr RegisterEventSource ()
 
 		public override void DisableNotification ()
 		{
-			if (_notifyResetEvent != null) {
-				_notifyResetEvent.Close ();
-				_notifyResetEvent = null;
+			lock (_eventLock) {
+				if (_notifyResetEvent != null) {
+					_notifyResetEvent.Close ();
+					_notifyResetEvent = null;
+				}
 			}
 
-			if (_notifyThread != null) {
-				if (_notifyThread.ThreadState == System.Threading.ThreadState.Running)
-					_notifyThread.Abort ();
-				_notifyThread = null;
-			}
+			_notifyThread = null;
+			
 		}
 
 		public override void EnableNotification ()
 		{
-			_notifyResetEvent = new ManualResetEvent (false);
-			_lastEntryWritten = OldestEventLogEntry + EntryCount;
-			if (PInvoke.NotifyChangeEventLog (ReadHandle, _notifyResetEvent.Handle) == 0)
-				throw new InvalidOperationException (string.Format (
-					CultureInfo.InvariantCulture, "Unable to receive notifications"
-					+ " for log '{0}' on computer '{1}'.", CoreEventLog.GetLogName (),
-					CoreEventLog.MachineName), new Win32Exception ());
-			_notifyThread = new Thread (new ThreadStart (NotifyEventThread));
-			_notifyThread.IsBackground = true;
-			_notifyThread.Start ();
+			lock (_eventLock) {
+				_notifyResetEvent = new ManualResetEvent (false);
+				_lastEntryWritten = OldestEventLogEntry + EntryCount;
+				if (PInvoke.NotifyChangeEventLog (ReadHandle, _notifyResetEvent.Handle) == 0)
+					throw new InvalidOperationException (string.Format (
+						CultureInfo.InvariantCulture, "Unable to receive notifications"
+						+ " for log '{0}' on computer '{1}'.", CoreEventLog.GetLogName (),
+						CoreEventLog.MachineName), new Win32Exception ());
+				_notifyThread = new Thread (() => NotifyEventThread (_notifyResetEvent));
+				_notifyThread.IsBackground = true;
+				_notifyThread.Start ();
```

Comment:
It's different, previous implementation had only 1 _notifyResetEvent and you have 1 per thread now.

I think you could return if _notifyResetEvent != null.